File: LeetCode Solutions/February LeetCoding Challenge/Search a 2D Matrix II.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row = 0, column = matrix[0].size() - 1;
        while(row < matrix.size() && column >= 0){
            if(matrix[row][column] > target){
                column--;
            }
            else if(matrix[row][column] < target){
                row++;
            } else{
                return true;
            }
        }
        return false;
    }
};

File: LeetCode Solutions/LeetCoding Challenge/February 2021 LeetCoding Challenge/Search a 2D Matrix II/Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row, column = 0, len(matrix[0]) - 1
        while (row < len(matrix) and column >= 0):
            if (matrix[row][column] > target):
                column -= 1
            elif (matrix[row][column] < target):
                row += 1
            else:
                return True
        return False